agent/connect/ca/provider_vault.go

```diff
@@ -66,9 +77,75 @@ func (v *VaultProvider) Configure(cfg ProviderConfig) error {
 	v.clusterID = cfg.ClusterID
 	v.spiffeID = connect.SpiffeIDSigningForCluster(&structs.CAConfiguration{ClusterID: v.clusterID})
 
+	// Look up the token to see if we can auto-renew its lease.
+	secret, err := client.Auth().Token().Lookup(config.Token)
+	if err != nil {
+		return err
+	}
+	var token struct {
+		Renewable bool
+		TTL       int
+	}
+	if err := mapstructure.Decode(secret.Data, &token); err != nil {
+		return err
+	}
+
+	// Set up a renewer to renew the token automatically, if supported.
+	if token.Renewable {
+		renewer, err := client.NewRenewer(&vaultapi.RenewerInput{
+			Secret: &vaultapi.Secret{
+				Auth: &vaultapi.SecretAuth{
+					ClientToken:   config.Token,
+					Renewable:     token.Renewable,
+					LeaseDuration: secret.LeaseDuration,
+				},
+			},
+			Increment: token.TTL,
+		})
+		if err != nil {
+			return fmt.Errorf("Error beginning Vault provider token renewal: %v", err)
+		}
+
+		ctx, cancel := context.WithCancel(context.TODO())
+		v.shutdown = cancel
+		go v.renewToken(ctx, renewer)
+	}
+
 	return nil
 }
 
+// renewToken uses a vaultapi.Renewer to repeatedly renew our token's lease.
+func (v *VaultProvider) renewToken(ctx context.Context, renewer *vaultapi.Renewer) {
+	go renewer.Renew()
+	defer renewer.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+
+		case err := <-renewer.DoneCh():
+			if err != nil {
+				v.logger.Error(fmt.Sprintf("Error renewing token for Vault provider: %v", err))
```

Comment:
During the hclog migration, these sorts of log statements became:
```suggestion
				v.logger.Error("Error renewing token for Vault provider", "error", err)
```